apps/test/unit/templates/studioHomepages/CourseCardTest.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {expect} from '../../../util/configuredChai';
+import CourseCard from '@cdo/apps/templates/studioHomepages/CourseCard';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+
+const testCourse = {
+  title: "Course 1",
+  description: "Start with Course 1 for early readers. Students will create computer programs that will help them learn to collaborate with others, develop problem-solving skills, and persist through difficult tasks. By the end of this course, students create their very own custom game or story that they can share. Recommended for grades K-1.",
+  link: "https://studio.code.org/s/course1",
+};
+
+describe('CourseCard', () => {
+  it('renders a card-shaped link', () => {
+    const wrapper = shallow(
+      <CourseCard
+        title={testCourse.title}
+        description={testCourse.description}
+        link={testCourse.link}
+        isRtl={false}
+      />
+    );
+    expect(wrapper).to.containMatchingElement(
+      <a href={testCourse.link}>
+        <img/>
+        <div>
+          {testCourse.title}
+        </div>
+        <div>
+          {testCourse.description}
+          <div>
+            <h3>
+              View course
+            </h3>
+            <FontAwesome icon="chevron-right"/>
+          </div>
+        </div>
+      </a>
+    );
+  });
+
+  it('can render in RTL mode', () => {
+    const wrapper = shallow(
+      <CourseCard
+        title={testCourse.title}
+        description={testCourse.description}
+        link={testCourse.link}
+        isRtl={true}
+      />
+    );
+    expect(wrapper).to.containMatchingElement(
+      <a href={testCourse.link}>
+        <img/>
+        <div>
+          {testCourse.title}
+        </div>
```

Comment:
Better documentation is found with [Enzyme's `containsMatchingElement` method](http://airbnb.io/enzyme/docs/api/ShallowWrapper/containsMatchingElement.html).

There's a predictable set of rules for matching, which generally let you leave unwanted detail out of your test case.

**Props in the expectation _must_ be present in the wrapper**
```jsx
const wrapper = shallow(<div id="abc"/>);
expect(wrapper).not.to.containMatchingElement(<div id="abc" className="123"/>);
```

**Props in the wrapper may be omitted from the expectation**
```jsx
const wrapper = shallow(<div id="abc" className="123"/>);
expect(wrapper).to.containMatchingElement(<div id="abc"/>);
expect(wrapper).to.containMatchingElement(<div/>);
```

**Object and array props are compared by deep equality, not strict equality**
```jsx
const wrapper = shallow(<div style={{color: 'red', top: 10}}/>);
expect(wrapper).to.containMatchingElement(<div style={{color: 'red', top: 10}}/>);
expect(wrapper).not.to.containMatchingElement(<div style={{color: 'red', top: 9}}/>);

const wrapper2 = shallow(<div data-row={[1, 2]}/>);
expect(wrapper2).to.containMatchingElement(<div data-row={[1, 2]}/>);
expect(wrapper2).not.to.containMatchingElement(<div data-row={[3]}/>);
```

**Children must match**
The `children` prop is a special case of the above rules - the _presence_ of children must match both ways, using this definition of match recursively.

For instance, children can also omit props: 
```jsx
const wrapper = shallow(
  <div>
    <a href="#">Link</a>
    <p className="description">Text</p>
  </div>
);
expect(wrapper).to.containMatchingElement(
  <div>
    <a>Link</a>
    <p>Text</p>
  </div>
);
```
Extra children in either direction fail the assertion.
```jsx
const wrapper = shallow(
  <div>
    <a href="#">Link</a>
  </div>
);
expect(wrapper).not.to.containMatchingElement(<div/>);
expect(wrapper).not.to.containMatchingElement(
  <div>
    <a href="#">Link</a>
    <p>Text</p>
  </div>
);
```
Also, text nodes count as children which _must_ be present and match.
```jsx
const wrapper = shallow(
  <div>
    <a href="#">Link</a>
  </div>
);
expect(wrapper).not.to.containMatchingElement(
  <div>
    <a/>
  </div>
);
```